src/views/Sales/SalesAdd.vue

```diff
@@ -169,35 +149,60 @@ const isFormValid = async () => {
 };
 
 const isProductsValidated = () => {
-  if (productsForSale.value.length <= 0) {
-    snackbarProvider.showErrorSnackbar("Please select a product!");
+  const isProductsValid = sellObject.products.value;
+  const isResourcesValid = sellObject.resources.value;
+
+  if (!isProductsValid && !isResourcesValid) {
+    snackbarProvider.showErrorSnackbar(
+      "Please select a product or a resource!"
+    );
     return false;
   }
   return true;
 };
 
 const isDateValidated = () => {
-  if (formattedDate.value.length <= 0) {
+  const isDateValid = sellObject.date;
+
+  if (isDateValid.length <= 0) {
```

Comment:
I believe it should just work if we do
```suggestion
  if (!saleDate) {
```

Reply:
it does not, because in the saleObject, date is defined, its checking if its defined or not